crates/services/executor/src/executor.rs

```diff
@@ -1030,148 +1080,342 @@ where
+
+        let coinbase_id = checked_mint.id();
+        let (mut mint, _) = checked_mint.into();
+
+        if mint.input_contract().contract_id == ContractId::zeroed() {
+            Self::verify_mint_for_empty_contract(&mint)?;
+        } else {
+            Self::check_mint_amount(&mint, execution_data.coinbase)?;
+            Self::check_gas_price(&mint, gas_price)?;
 
-            self.persist_output_utxos(
-                block_height,
+            let input = mint.input_contract().clone();
+            let mut inputs = [Input::Contract(input)];
+
+            if self.options.utxo_validation {
+                self.verify_inputs_exist(
+                    block_st_transaction,
+                    inputs.as_mut_slice(),
+                    header.da_height,
+                )?;
+            }
+
+            self.compute_inputs(inputs.as_mut_slice(), block_st_transaction)?;
+
+            let (input, output) = self.execute_mint_with_vm(
+                &header,
+                coinbase_contract_id,
                 execution_data,
-                &coinbase_id,
                 block_st_transaction,
-                inputs.as_slice(),
-                outputs.as_slice(),
+                &coinbase_id,
+                &mut mint,
+                &mut inputs,
             )?;
-            self.compute_state_of_not_utxo_outputs(
-                outputs.as_mut_slice(),
-                inputs.as_slice(),
+
+            *mint.input_contract_mut() = input;
+            *mint.output_contract_mut() = output;
+        }
+
+        Self::store_mint_tx(mint, execution_data, coinbase_id, block_st_transaction)
+    }
+
+    fn validate_mint<T>(
+        &self,
+        checked_mint: Checked<Mint>,
+        header: &PartialBlockHeader,
+        coinbase_contract_id: ContractId,
+        gas_price: Word,
+        execution_data: &mut ExecutionData,
+        block_st_transaction: &mut StorageTransaction<T>,
+    ) -> ExecutorResult<Transaction>
+    where
+        T: KeyValueInspect<Column = Column>,
+    {
+        execution_data.found_mint = true;
+
+        Self::check_mint_has_expected_index(&checked_mint, execution_data)?;
+
+        let coinbase_id = checked_mint.id();
+        let (mut mint, _) = checked_mint.into();
+
+        if mint.input_contract().contract_id == ContractId::zeroed() {
+            Self::verify_mint_for_empty_contract(&mint)?;
+        } else {
+            Self::check_mint_amount(&mint, execution_data.coinbase)?;
+            Self::check_gas_price(&mint, gas_price)?;
+
+            let input = mint.input_contract().clone();
+            let mut inputs = [Input::Contract(input)];
+
+            if self.options.utxo_validation {
+                self.verify_inputs_exist(
+                    block_st_transaction,
+                    inputs.as_mut_slice(),
+                    header.da_height,
+                )?;
+            }
+
+            self.validate_inputs_state(
```

Comment:
I had this missing for a while and the tests were still passing. Should we add coverage for this or is it unnecessary?